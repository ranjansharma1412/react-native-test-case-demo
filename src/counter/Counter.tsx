import React, { useState } from 'react'
import { View, Text, Button } from 'react-native'

const Counter = ({ initialCount }: { initialCount: number }) => {
    const [count, setCount] = useState(initialCount)

    const incrementCount = () => {
        setCount(count + 1);
    }
    const decrementCount = () => {
        setCount(count - 1);
    }

    return (
        <View style={{ flex: 1, justifyContent: 'center', paddingHorizontal: 20 }}>
            <Text >Count: {count}</Text>
            <View style={{ marginTop: 20 }}>
                <Button onPress={incrementCount} title='Increment' />
            </View>
            <View style={{ marginTop: 20 }}>
                <Button onPress={decrementCount} title='Decrement' />
            </View>
        </View>
    )
}

export default Counter