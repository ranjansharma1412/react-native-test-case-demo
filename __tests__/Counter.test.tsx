
import { it, describe, expect } from '@jest/globals';
// import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react-native';
import Counter from '../src/counter/Counter';


describe(Counter, () => {
    it("counter display correct initial count", () => {
        render(<Counter initialCount={0} />);

        const countDisplay = Number(screen.getByTestId('count-display').props.children);
        expect(countDisplay).toEqual(0);

        // increment
        const incrementButton = screen.getByTestId('increment-button');
        fireEvent.press(screen.getByTestId('increment-button'));
        const countDisplay2 = Number(screen.getByTestId('count-display').props.children);
        expect(countDisplay2).toEqual(1);

        //decrement
        const decrementButton = screen.getByTestId('decrement-button');
        fireEvent.press(decrementButton);
        const countDisplay3 = Number(screen.getByTestId('count-display').props.children);
        expect(countDisplay3).toEqual(0);

        expect(screen.toJSON()).toMatchSnapshot()
    })
})