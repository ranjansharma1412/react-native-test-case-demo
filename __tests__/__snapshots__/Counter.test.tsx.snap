// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter counter display correct initial count 1`] = `
<View
  style={
    {
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
    }
  }
>
  <Text
    testID="count-display"
  >
    0
  </Text>
  <View
    style={
      {
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="increment-button"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Increment
        </Text>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 20,
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="decrement-button"
    >
      <View
        style={
          [
            {},
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Decrement
        </Text>
      </View>
    </View>
  </View>
</View>
`;
